packages/cinterop/src/commonMain/kotlin/io/realm/internal/interop/RealmInterop.kt

```diff
@@ -52,6 +52,7 @@ expect object RealmInterop {
     fun realm_config_set_max_number_of_active_versions(config: NativePointer, maxNumberOfVersions: Long)
     fun realm_config_set_encryption_key(config: NativePointer, encryptionKey: ByteArray)
     fun realm_config_get_encryption_key(config: NativePointer): ByteArray?
+    fun realm_config_set_should_compact_on_launch_function(config: NativePointer, callback: (Long, Long) -> Boolean)
```

Comment:
Maybe make a typemap for the callback type?

Reply:
If it is made a functional interface instead it would simplify the JNI code as it does not need the arguents to be boxed.